docs/howto/garbage-collection.md

```diff
@@ -0,0 +1,254 @@
+---
+layout: default
+title: Garbage Collection
+description: Clean up unnecessary objects using the garbage collection feature in lakeFS.
+parent: Garbage Collection
+grand_parent: How-To
+nav_order: 1
+has_children: false
+redirect_from:
+  - /reference/garbage-collection.html
+  - /howto/garbage-collection-index.html
+---
+
+[lakeFS Cloud](https://lakefs.cloud) users enjoy a managed garbage collection service, and do not need to run this Spark program.
+{: .tip }
+
+# Garbage Collection
+
+By default, lakeFS keeps all your objects forever. This allows you to travel back in time to previous versions of your data.
+However, sometimes you may want to remove the objects from the underlying storage completely.
+Reasons for this include cost-reduction and privacy policies.
+
+The garbage collection job is a Spark program that removes the following from the underlying storage:
+1. _Uncommitted objects_ that are no longer accessible
+   * For example, objects deleted before ever being committed.
+2. _Committed objects_ that are considered expired according to retention rules you define.
+
+{% include toc.html %}
+
+## Understanding retention rules
+
+{: .note }
+Without retention rules, only inaccessible uncommitted objects will be removed by the job.
+
+For every branch, the GC job retains deleted objects for the number of days defined for the branch.
+In the absence of a branch-specific rule, the default rule for the repository is used.
+If an object is present in more than one branch ancestry, it's retained according to the rule with the largest number of days between those branches.
+That is, it's removed only after the retention period has ended for all relevant branches.
+
+Example GC rules for a repository:
+```json
+{
+  "default_retention_days": 14,
+  "branches": [
+    {"branch_id": "main", "retention_days": 21},
+    {"branch_id": "dev", "retention_days": 7}
+  ]
+}
+```
+
+In the above example, objects will be retained for 14 days after deletion by default.
```

Comment:
I wonder if we should add the overridden scenario, i.e. paths that were overridden and therefore the objects that previously represented them were deleted

Reply:
Added